### PR TITLE
fix: update private changed for liveview 0.19.0 (closes: #8)

### LINKED ISSUE #8: Broken in LiveView 0.19.0
Calling `Favicon.set_attr(socket, "href", favicon(icon))` (which uses `Head.push(socket, "link[rel*='icon']", :set, attr, value)`) in LiveView 0.19.0 results in a key error:
```elixir
key :__changed__ not found in: %{__temp__: %{push_events: [["hd", %{c: [["f", [[:i, "h", "i"]]]]}]]},
  assign_new: {%{current_user: ...}
```

Digging through 0.19.0's changes now, and it looks like they changed from using `priv.__changed__` to `priv.__temp__`:

https://github.com/phoenixframework/phoenix_live_view/commit/c67d6dc7cfaf7586246d5b7091061b10296ef30b

### PR DESCRIPTION
**Pull request by dfalling**
This resolves the incompatibility introduced in LiveView 0.19.0.

Closes #8

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -36,7 +36,7 @@ defmodule Phx.Live.Head.MixProject do
     [
       {:phoenix, ">= 1.6.0"},
       {:phoenix_html, ">= 3.1.0"},
-      {:phoenix_live_view, ">= 0.17.0"},
+      {:phoenix_live_view, ">= 0.19.0"},
```
Comment (dfalling) [inline mix.exs]: This version will not work in previous versions of LiveView, so I had to bump the minimum version up.

File: lib/live_head.ex [hunk 2]
```diff
@@ -164,7 +164,7 @@ defmodule Phx.Live.Head do
     #  head events or we replace the whole list of events with our mapped variant
     #  including merged head events
     if merged? do
-      put_in(socket.private.__changed__.push_events, events)
+      put_in(socket.private.__temp__.push_events, events)
```
Comment (BartOtten) [inline lib/live_head.ex]: I think we can make the fix without needing a version bump by detecting which field is supported. Could you have a try with something like:

```elixir
cond do
  # Phoenix LiveView < 0.19
  merged? && Map.has_key?(socket.private, :__changed__) ->
    put_in(socket.private.__changed__.push_events, events)
  
  # Phoenix LiveView >= 0.19
  merged? -> 
    put_in(socket.private.__temp__.push_events, events)
    
  true -> 
    push_event(socket, "hd", %{c: [[query, [change]]]})
end
```

**Review by BartOtten — CHANGES_REQUESTED** [review]
Thanks for not only reporting the issue but also create a PR to resolve it. Have included a suggestion to not need a version bump.

File: lib/live_head.ex (hunk 2 above)
Comment (dfalling) [reply lib/live_head.ex]: Nice, that's much better than restricting the LiveView version. I tweaked the comments a bit as well.

**Review by BartOtten — DISMISSED** [review]

**Comment by dfalling** [comment]
The new logic pushed us over Credo's complexity limit. Any preference on how to resolve this? Maybe splitting the logic to find the `{events, merged?}` out to a separate helper function?

**Comment by BartOtten** [comment]
Hi @dfalling,

I agree with you it seems we can resolve it by extracting some private helper functions.
    
```
  defp event_reducer(["hd", %{c: changes}], _merged?, query, change),
     do:  {["hd", %{c: push_or_merge_head_change(changes, query, change)}], true}

  defp event_reducer(event, merged?, _query, _change), 
      do: {event, merged?}

```
And while we are at it

```
  defp put_event(socket, events) when is_map_key(socket.private, :__changed__)
  defp put_event(socket, events) when is_map_key(socket.private, :__temp__)
```
 
This way there can be a simple if/else in push_or_merge, instead of the `cond`.

Could you give it a try? :)

**Review by BartOtten — APPROVED** [review]

**Comment by BartOtten** [comment]
@dfalling Thanks for your contribution! It lifts my spirit :)

**Comment by dfalling** [comment]
Sure thing, thanks for the great lib!

**Pull request merged**